### PR TITLE
Replace swc with ts-jest typecheck unit tests

### PR DESCRIPTION
**Pull request by birkskyum**
I have installed [ts-jest](https://kulshekhar.github.io/ts-jest/), so we now have type checked tests.

### CODE REVIEW

**Comment by HarelM** [comment]
This is a great start, thanks!
The question is if we want to run the tests with swc to get better petformance and tun in parallel the tsc simply to test for type correctness?
It's just a thought...

**Comment by birkskyum** [comment]
> This is a great start, thanks! The question is if we want to run the tests with swc to get better petformance and tun in parallel the tsc simply to test for type correctness? It's just a thought...

I think we should stick with ts-jest at least temporary until we have figured out how to get compatible with new ts-node and node 18, as swc were making the transition more difficult. We can always enable it again afterwards.

**Comment by birkskyum** [comment]
it takes me 20 sec to run the unit tests with ts-jest, so it's not back to the 90 sec that we had with babel etc.

**Comment by birkskyum** [comment]
To get this through ci, I have added some ts-nocheck that we can then gradually remove to get everything typed.

File: build/generate-style-spec.ts [hunk 1]
```diff
@@ -115,7 +115,9 @@ function layerType(key) {
 const layerTypes = Object.keys(spec.layer.type.values);
 
 fs.writeFileSync('src/style-spec/types.g.ts',
-    `// Generated code; do not edit. Edit build/generate-style-spec.ts instead.
+    `// @ts-nocheck
```
Comment (HarelM) [inline build/generate-style-spec.ts:118]: What's wrong with the generated code? It passes regular build with rollup, doesn't it?

File: tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,6 @@
 {
-  "ts-node": {
-    "swc": true
-  },
   "compilerOptions": {
-    "allowJs": false,
+    "allowJs": true,
```
Comment (HarelM) [inline tsconfig.json:3]: Where are we using non-typesccript files that this is needed?

File: build/generate-style-spec.ts (hunk 1 above)
Comment (birkskyum) [reply build/generate-style-spec.ts]: Good catch. I got a type error before, but I can't replicate it now, so I rolled back this change.

File: tsconfig.json (hunk 2 above)
Comment (birkskyum) [reply tsconfig.json:3]: Many of the tests break if it's set to false, because it picks up .js files from libraries we use like tinysdf etc.

**Comment by HarelM** [comment]
Overall looks good! Thanks! 
Looks like not a lot of files are causing the problems so I hope it won't take too long to fix them.

File: tsconfig.json (hunk 2 above)
Comment (HarelM) [reply tsconfig.json:3]: Ah, thanks! I didn't know that.

**Comment by birkskyum** [comment]
> Overall looks good! Thanks! Looks like not a lot of files are causing the problems so I hope it won't take too long to fix them.

Great. I was actually thinking of merging it as is so it isn't blocking everything else, and then chip away the nocheck lines little by little.

**Comment by HarelM** [comment]
Sounds like a good plan.
The only comment I have is that if this is merged #1391 should remain open until all comments are removed.

**Review by HarelM — APPROVED** [review]

**Pull request merged**